pkg/verifier/safe_delete.go

```diff
@@ -14,10 +15,14 @@ import (
 	"github.com/pkg/errors"
 )
 
-// SafeDelete moves block to backup bucket and if succeeded, removes it from source bucket.
-// It returns error if block dir already exists in backup bucket (blocks should be immutable) or any
-// of the operation fails.
-func SafeDelete(ctx context.Context, logger log.Logger, bkt objstore.Bucket, backupBkt objstore.Bucket, id ulid.ULID) error {
+// SafeDelete moves a TSDB block to a backup bucket and, on success, removes
+// it from the source bucket.  It returns error if block dir already exists in
+// the backup bucket (blocks should be immutable) or if any of the operations
+// fail.  When useExisting is true do not download the block but use the
+// previously downloaded block found in tempdir to avoid repeated downloads.
+// If tempdir is an empty string a unique temporary directory is created for
+// scratch space, otherwise the given directory is used.
+func SafeDelete(ctx context.Context, logger log.Logger, bkt objstore.Bucket, backupBkt objstore.Bucket, id ulid.ULID, useExisting bool, tempdir string) error {
```

Comment:
sorry for nitpicking on this, but can we name this `useExistingTempdir`? I realize that's long, but we'll understand in 3 months what this means that way :)

Reply:
Not a problem.  Fixed.